lib/data_cycle/short_update.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "#{Rails.root}/lib/data_cycle/batch_update_logging"
+require "#{Rails.root}/lib/course_revision_updater"
+require "#{Rails.root}/lib/assignment_updater"
+require "#{Rails.root}/lib/importers/revision_score_importer"
+require "#{Rails.root}/lib/importers/plagiabot_importer"
+require "#{Rails.root}/lib/importers/upload_importer"
+require "#{Rails.root}/lib/importers/view_importer"
+require "#{Rails.root}/lib/importers/rating_importer"
+require "#{Rails.root}/lib/data_cycle/cache_updater"
+require "#{Rails.root}/lib/data_cycle/update_cycle_alert_generator"
+require "#{Rails.root}/lib/student_greeting_checker"
+
+# Executes all the steps of 'update_constantly' data import task
+class ShortUpdate
+  include BatchUpdateLogging
+  include CacheUpdater
+  include UpdateCycleAlertGenerator
+
+  def initialize
+    setup_logger
+    set_courses_to_update
+    return if updates_paused?
+    return if conflicting_updates_running?
+
+    run_update_with_pid_files(:constant)
+  end
+
+  private
+
+  def set_courses_to_update
+    @courses = Course.ready_for_short_update.to_a
+    log_message "Ready to update #{@courses.count} courses"
+  end
+
+  def run_update
+    log_start_of_update 'Short update tasks are beginning.'
+    update_revisions_and_articles
+    update_new_article_views unless ENV['no_views'] == 'true'
+    update_new_article_ratings
+    import_uploads_for_needs_update_editathons
```

Comment:
Do we need the uploads and to format the categories for this update? Normally this will also happen in the daily update, so it's not that relevant. Anyway, I reduced also the scope of courses that should be updated in the methods definitions.